src/Illuminate/Http/Resources/Json/JsonResource.php

```diff
@@ -230,4 +269,80 @@ public function jsonSerialize()
     {
         return $this->resolve(Container::getInstance()->make('request'));
     }
+
+    /**
+     * Get a new resource instance for the given model name.
+     *
+     * @param string $modelName
+     * @param mixed ...$parameters
+     * @return static
+     */
+    public static function resourceForModel(string $modelName, ...$parameters)
+    {
+        $resource = static::resolveResourceName($modelName);
+
+        return $resource::make(...$parameters);
+    }
+
+    /**
+     * Specify the callback that should be invoked to guess factory names based on dynamic relationship names.
+     *
+     * @param  callable  $callback
+     * @return void
+     */
+    public static function guessResourceNamesUsing(callable $callback)
+    {
+        static::$resourceNameResolver = $callback;
+    }
+
+    /**
+     * Specify the default namespace that contains the application's API resources.
+     *
+     * @param  string  $namespace
+     * @return void
+     */
+    public static function useNamespace(string $namespace)
+    {
+        static::$namespace = $namespace;
+    }
+
+    /**
+     * Get the resource name for the given model name.
+     *
+     * @param  string  $modelName
+     * @return string
+     */
+    public static function resolveResourceName(string $modelName)
+    {
+        $resolver = static::$resourceNameResolver ?: function (string $modelName) {
+            $appNamespace = static::appNamespace();
+
+            $modelName = Str::startsWith($modelName, $appNamespace.'Models\\')
+                ? Str::after($modelName, $appNamespace.'Models\\')
+                : Str::after($modelName, $appNamespace);
+            $resourceName = static::$namespace.$modelName;
+
+            return class_exists($resourceName)
+                ? $resourceName
+                : $resourceName.'Resource';
+        };
+
+        return $resolver($modelName);
+    }
+
+    /**
+     * Get the application namespace for the application.
+     *
+     * @return string
+     */
+    protected static function appNamespace()
+    {
+        try {
+            return Container::getInstance()
+                ->make(Application::class)
+                ->getNamespace();
+        } catch (Throwable $e) {
+            return 'App\\';
```

Comment:
same issue

Reply:
I lifted this logic from the factory... factory. What's your suggestion on how to achieve this? I would hate to have it be an extra setup in a service provider

Reply:
I guess this spot is fine as it was how @taylorotwell addressed usage outside of the framework for PR #35204 

See his comment here:

https://github.com/laravel/framework/pull/35204#discussion_r522987927

And solution here:

https://github.com/laravel/framework/commit/8bf80de4b96b507fcc74a3e217d834882fefa67a#diff-14a15185e5dabb6621c27ded8fda4891a78b857b0872e46d52c3dacce6457564

But I think on the other spot (`protected static $namespace = 'App\\Http\\Resources\\';`) you will need to come up with another solution.

Perhaps adding a resolver callback that is set on a Service Provider similar to what is done for other Illuminate components (such as the Event dispatcher's `setQueueResolver` on `Illuminate\Events\EventServiceProvider`)

Reply:
Great point. I'll make some updates to that